src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -156,21 +164,66 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                 {
                     try
                     {
-                        // Don't use the individual CT here, this is a shared operation that shouldn't be affected by an individual's cancellation.
-                        // The transport should have it's own timeouts, etc..
-                        var configuration = await _configRetriever.GetConfigurationAsync(MetadataAddress, _docRetriever, CancellationToken.None).ConfigureAwait(false);
-                        if (_configValidator != null)
+                        T configuration = null;
+                        if (DistributedConfigurationManager != null && !_skipDistributedConfigurationManager)
                         {
-                            ConfigurationValidationResult result = _configValidator.Validate(configuration);
-                            if (!result.Succeeded)
-                                throw LogHelper.LogExceptionMessage(new InvalidConfigurationException(LogHelper.FormatInvariant(LogMessages.IDX20810, result.ErrorMessage)));
+                            try
+                            {
+                                configuration = await DistributedConfigurationManager.GetConfigurationAsync(MetadataAddress, s_distributedConfigurationOptions, CancellationToken.None).ConfigureAwait(false);
+                                if (_configValidator != null)
+                                {
+                                    ConfigurationValidationResult result = _configValidator.Validate(configuration);
+                                    if (!result.Succeeded)
+                                    {
+                                        configuration = null;
+                                        LogHelper.LogExceptionMessage(new InvalidConfigurationException(LogHelper.FormatInvariant(LogMessages.IDX20812, result.ErrorMessage)));
```

Comment:
I updated the log message so that it is different from that of the l1 cache. Is `InvalidConfigurationException` okay or do we want a new custom exception?